share/spack/setup-env.sh

```diff
@@ -212,7 +212,8 @@ _spack_determine_shell() {
         # If procfs is present this seems a more reliable
         # way to detect the current shell
         _sp_exe=$(readlink /proc/$$/exe)
-        basename ${_sp_exe}
+        # Shell may contain number, like zsh5 instead of zsh
+        basename ${_sp_exe} | tr -d '0123456789'
```

Comment:
I wonder if this would be fragile. Also I noticed that `zsh5` seems to be a deprecated command to invoke `zsh`:
```console
$ zsh5 --version

    ####
    ####  /bin/zsh5 is deprecated, please switch to /bin/zsh (or /usr/bin/zsh)
    ####  This command (/bin/zsh5) will disappear eventually!
    ####

zsh 5.4.2 (x86_64-ubuntu-linux-gnu)
```
Considering the size of the executables it's probably a wrapper around it:
```console
$ ls -larth /bin/zsh*
-rwxr-xr-x 1 root root  843 dic 30  2017 /bin/zsh5
-rwxr-xr-x 1 root root 813K set 10  2018 /bin/zsh
```

Reply:
Why would it be fragile?

Reply:
> The procfs tests were ~unreliable~ too reliable

LOL

> Why would it be fragile?

Maybe I am overthinking it, but removing numbers fits this particular case and nobody ensure us that some `exe` won't have in the future numbers as part of its name that we have to retain. It's also true that we can adjust to that situation when and if it happens.

Reply:
I think the number of exe we have to detect is currently pretty limited to bash, zsh, dash, and sh. I don’t think we’ll have to worry about numbers in the near future.

Reply:
> I don’t think we’ll have to worry about numbers in the near future.

Fair enough